python/grass/jupyter/interact_display.py

```diff
@@ -0,0 +1,158 @@
+                       e=self._extent["east"],
+                       w=self._extent["west"],
+                       env=self._env
+                      )
+        
+        # Get Center of tmp GRASS region
+        center = gs.parse_command("g.region", flags="cg", env=self._env)
+        center = (float(center["center_northing"]), float(center["center_easting"]))
+
+        # Create Folium Map 
+        self.map = folium.Map(
+            width=self.width,
+            height=self.height,
+            location=center,
+            tiles="cartodbpositron",
+        )
+
+        
+    def _get_rc(self, env=None):
+        rc = {}
+        
+        if env:
+            rcfile_location = env["GISRC"]
+        else:
+            rcfile_location = os.environ["GISRC"]
+            
+        with open(rcfile_location) as f:
+            for line in f:
+                entry = line.strip("\n").split(": ")
+                rc[entry[0]] = entry[1]
+        return rc
+    
+    
+
+    def _convert_coordinates(self, x, y, proj_in):
+        """This function reprojects coordinates to WGS84, the required
+        projection for folium.
+
+        Coordinates should be a string with x and y values separated
+        by a comma.
+
+        proj_in is a proj4 string, for example, the output of g.region
+        with the `g` flag."""
+        # Reformat input
+        coordinates = f"{x}, {y}"
+        
+        # Reproject coordinates
+        coords_folium = gs.read_command(
+            "m.proj",
+            coordinates=coordinates,
+            proj_in=proj_in,
+            separator="comma",
+            flags="do",
+        )
+
+        # Reformat from string to array
+        coords_folium = coords_folium.strip()  # Remove '\n' at end of string
+        coords_folium = coords_folium.split(",")  # Split on comma
+        coords_folium = [float(value) for value in coords_folium]  # Convert to floats
+
+        return coords_folium[1], coords_folium[0]  # Return Lat and Lon
+
+    def _convert_extent(self, env=None):
+        """This function returns the bounding box of the current region
+        in WGS84, the required projection for folium"""
+
+        # Get proj of current GRASS region
+        proj = gs.read_command("g.proj", flags="jf", env=env)
+
+        # Get extent
+        extent = gs.parse_command("g.region", flags="g", env=env)
+
+        # Convert extent to EPSG:3857, required projection for Folium
+        north, east = self._convert_coordinates(extent["e"], extent["n"], proj)
+        south, west = self._convert_coordinates(extent["w"], extent["s"], proj)
+
+        extent = {'north': north, 'south': south, 'east': east, 'west': west}
+
+        return extent
+    
+    def _folium_bounding_box(self, extent):
+        """Reformats extent into bounding box to pass to folium"""
+        return [[extent['north'], extent['west']], [extent['south'], extent['east']]]
+    
+    def add_vector(self, vector):
```

Comment:
I'm generally see `add_vector(vector="abc")` as a missed opportunity to document what the parameter is. In a general Python context, I would say `add_vector(name="abc")`. You could also say it should always be called as `add_vector("abc")` and leave your hands free for later change of the parameter name.

Reply:
To clarify, the missed opportunity in `add_vector(vector="abc")` is that you are saying twice you are dealing with a vector. It is just repeating the information. Actually calling the function with the keyword argument then feels repetitive. Defining the function as `add_vector(self, name)` adds some information about what the argument actually is (it is a name, so string, and not some complex object - although in theory we could support things like that, in Python `arg` is often used as an argument name in that case from what I have seen). On the other hand, name does not add that much info in the actual call `add_vector(name="abc")`, so perhaps using it just as `add_vector("abc")` makes more sense. You could argue that name is name of the layer as displayed in the legend I guess. If you say in documentation, never use keyword argument for the first argument, you can change the name any time without breaking your API. That's the Python side of things.

In GRASS GIS, the parameters always have names like in Python and in some contexts, it is okay to leave out the name of the first one (in Python you can _generally_ (not always) leave out any name). The current practice is that if there is one raster or vector map as input and there is no output, it is called _map_, it is called _input_ if there is also output(s), and it is called _vector_ when there is also raster and vice versa. This was called into question because the rules are unclear to some users (defining "no output" is one of the tricky parts). The non-interactive d.vect-like functions will use this GRASS parameter name, however, we are not bound by these names here and can pick what works best.

Reply:
Ah, ok, I understand better now. So, let's go with `add_vector(self, name)`? That would match both Python and GRASS convention since it is the first parameter (see 1st sentence, 2nd paragraph. And, I agree that adding `name=` doesn't add much information.

Reply:
Also, reading your comment above again, I see your point. I need to use `["name"]` from `gs.find_file` instead of from whatever the user provides so that it will work if the user provides `"abc@PERMANENT"` - that way you don't end up with a file called "tmp_abc@PERMANENT.json".

Reply:
> ...let's go with `add_vector(self, name)`? ...adding `name=` doesn't add much information.

Sounds good, but say in the documentation that "*name* a positional-only parameter" (as, for example, *x* is described in [documentation of float buildin](https://docs.python.org/3/library/functions.html#float). There is a way in [new Python versions](https://docs.python.org/3/reference/compound_stmts.html#function-definitions) to do that on language-level rather than just in the documentation, but we need to support older version too (you can see full info in 
[PEP 570 -- Python Positional-Only Parameters](https://www.python.org/dev/peps/pep-0570/)).

This way you are not making any promises about the name of the parameter and are saying that `add_vector("abc")` is the right way of using this method.

Reply:
Perhaps we can talk about this at our next meeting. I have some lingering confusion and am not sure if I've done the correct way. I think the way I have it written is a "positional-only parameter"? From reading those links, it's my understanding that, with the way it's currently written, the user couldn't use `add_vector(self, name="abc")` which inherently means it's a "positional-only parameter."

Reply:
Let's talk about it then. In the meantime, try running `add_vector(self, vector="abc")` yourself.